### PR TITLE
Task6 done

### LINKED ISSUE #6: Implement island search in Map class Task#6

### PR DESCRIPTION
**Pull request by iprice73**

### CODE REVIEW

File: shm/Island.hpp [hunk 1]
```diff
@@ -7,5 +7,6 @@ class Island {
     Coordinates position_;
 public:
     Island(size_t positionX, size_t positionY) : position_(positionX, positionY) {};
-    const Coordinates& getPosition() { return position_; };
-};
\ No newline at end of file
+    Coordinates& getPosition() { return position_; };
```
Comment (J-Cieplinski) [inline shm/Island.hpp]: This change needs to be reverted back to const.

File: shm/Map.cpp [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
 }
 
 void Map::addIsland(const Island& newIsland) {
     islandsLocations_.push_back(newIsland);
 }
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    auto island = std::find_if(islandsLocations_.begin(), islandsLocations_.end(),
+                               [&coordinate](auto& isld) {
+                                   return isld.Island::getPosition() == coordinate;
```
Comment (PierwszyKrytykRP) [inline shm/Map.cpp:16]: ~~If you want to update position of an Island create a setter. Getters shouldn't be used to set variables.~~

Scratch that, you are comparing. I read that wrong

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

**Review by J-Cieplinski — APPROVED** [review]

**Review by ArturGubala — APPROVED** [review]

File: shm/Map.cpp [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
 }
 
 void Map::addIsland(const Island& newIsland) {
     islandsLocations_.push_back(newIsland);
 }
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    auto island = std::find_if(islandsLocations_.begin(), islandsLocations_.end(),
+                               [&coordinate](const auto& isld) {
+                                   return isld.Island::getPosition() == coordinate;
```
Comment (niemarcin) [inline shm/Map.cpp:16]: Do we need to write Island:: ? `isld.getPosition();` is not enough? `isld` is an element from vector of Islands so it should be able to get to getPosition() method, I guess.

**Review by niemarcin — APPROVED** [review]
@LogarithmDerivative looks fine - I wrote just one question about `Island::` in a comment to code

File: shm/Map.cpp (hunk 3 above)
Comment (iprice73) [reply shm/Map.cpp:16]: I thought the same but we don't have Island.cpp where `Island::` should be, we have to implement it yet and I will refactor this to `isld.GetPosition()`

**Review by lisie-sprawy — APPROVED** [review]

**Pull request merged**